### ISSUE TITLE
automatically find the source of package

### ISSUE THREAD

**Issue by asukaminato0721** [SukkaW/nolyfill#112]
Since these packages are not likely to be directly included, most of them are included by deps.

Is it possible to find the chain of deps that includes the package, and do something like auto-pr to that deps to maximize the effect?

**Comment by asukaminato0721** [comment]
Using `npm ls`

```
├─┬ archiver-utils@5.0.2 extraneous -> ./node_modules/.pnpm/archiver-utils@5.0.2/node_modules/archiver-utils
│ └─┬ readable-stream@4.5.2 -> ./node_modules/.pnpm/readable-stream@4.5.2/node_modules/readable-stream
│   └─┬ events@3.3.0 -> ./node_modules/.pnpm/events@3.3.0/node_modules/events
│     └── isarray@1.0.0 deduped invalid: "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/events@3.3.0/node_modules/events -> ./node_modules/.pnpm/isarray@1.0.0/node_modules/isarray
├─┬ arraybuffer.prototype.slice@1.0.3 extraneous -> ./node_modules/.pnpm/arraybuffer.prototype.slice@1.0.3/node_modules/arraybuffer.prototype.slice
│ └─┬ es-abstract@1.23.3 -> ./node_modules/.pnpm/es-abstract@1.23.3/node_modules/es-abstract
│   ├─┬ available-typed-arrays@1.0.7 -> ./node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays
│   │ └── isarray@1.0.0 deduped invalid: "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/events@3.3.0/node_modules/events, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays -> ./node_modules/.pnpm/isarray@1.0.0/node_modules/isarray
│   └─┬ safe-array-concat@1.1.2 -> ./node_modules/.pnpm/safe-array-concat@1.1.2/node_modules/safe-array-concat
│     └── isarray@2.0.5 deduped -> ./node_modules/.pnpm/isarray@2.0.5/node_modules/isarray
├─┬ aws-sdk@2.1691.0 -> ./node_modules/.pnpm/aws-sdk@2.1691.0/node_modules/aws-sdk
│ └─┬ buffer@4.9.2 -> ./node_modules/.pnpm/buffer@4.9.2/node_modules/buffer
│   └── isarray@1.0.0 -> ./node_modules/.pnpm/isarray@1.0.0/node_modules/isarray
├─┬ is-array-buffer@3.0.4 extraneous -> ./node_modules/.pnpm/is-array-buffer@3.0.4/node_modules/is-array-buffer
│ └─┬ available-typed-arrays@1.0.7 -> ./node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays
│   └── isarray@1.0.0 invalid: "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays, "^2.0.5" from node_modules/.pnpm/events@3.3.0/node_modules/events, "^2.0.5" from node_modules/.pnpm/available-typed-arrays@1.0.7/node_modules/available-typed-arrays -> ./node_modules/.pnpm/isarray@1.0.0/node_modules/isarray
├─┬ lazystream@1.0.1 extraneous -> ./node_modules/.pnpm/lazystream@1.0.1/node_modules/lazystream
│ └─┬ readable-stream@2.3.8 -> ./node_modules/.pnpm/readable-stream@2.3.8/node_modules/readable-stream
│   └── isarray@1.0.0 -> ./node_modules/.pnpm/isarray@1.0.0/node_modules/isarray
```

**Comment by ljharb** [comment]
npm explain gives you a much clearer answer to that question.
